src/resource_pool_texture.cpp

```diff
@@ -0,0 +1,50 @@
+#include "resource_pool_texture.hpp"
+#include "util/log.hpp"
+
+namespace wr
+{
+
+	TexturePool::TexturePool(std::size_t size_in_mb)
+		: m_size_in_mb(size_in_mb)
+	{
+	}
+
+	//! Loads a texture
+	TextureHandle TexturePool::Load(std::string_view path, bool srgb)
+	{
+		std::size_t length = path.length();
+		std::string_view extension = path.substr(length - 3, 3);
+
+		Texture* texture;
+
+		if (extension == "png")
```

Comment:
Replace this with a proper wildcard match function.

Reply:
Discussed in person about the alternative. Solved.